resources/ts/filters/filter-dropdown.ts

```diff
@@ -1,51 +1,100 @@
 import submitForm from '../form/form-handler';
+import isReducedMotion from '../common/reduced-motion-detection';
 
 const DROPDOWN_SELECTOR = '.fau-dropdown';
 const DROPDOWN_TOGGLE_SELECTOR = '.fau-dropdown__toggle';
 const DROPDOWN_CONTENT_SELECTOR = '.fau-dropdown__content';
 const CLICKAWAY_WINDOW_WIDTH_THRESHOLD = 768; // close on click away only works on screen sizes above this value
 
-const closeDropDown = ( dropdown: HTMLElement ) => {
-	dropdown.setAttribute( 'aria-expanded', 'false' );
-	submitForm();
-};
+class Dropdown {
+	private dropdown: HTMLElement;
+	private toggle: HTMLElement;
+	private content: HTMLElement;
+	private checkboxes: HTMLInputElement[];
+	private initialState: boolean[] = [];
 
-const toggleDropdown = ( dropdown: HTMLElement ) => {
-	const isAriaExpanded = dropdown.getAttribute( 'aria-expanded' ) === 'true';
+	constructor( dropdown: HTMLElement ) {
+		this.dropdown = dropdown;
+		this.toggle = dropdown.querySelector( DROPDOWN_TOGGLE_SELECTOR )!;
+		this.content = dropdown.querySelector( DROPDOWN_CONTENT_SELECTOR )!;
+		this.checkboxes = Array.from(
+			this.content.querySelectorAll( 'input[type="checkbox"]' )
+		);
 
-	dropdown.setAttribute( 'aria-expanded', isAriaExpanded ? 'false' : 'true' );
+		this.init();
+	}
 
-	if ( isAriaExpanded ) {
-		submitForm();
+	private recordInitialState() {
+		this.initialState = this.checkboxes.map(
+			( checkbox ) => checkbox.checked
+		);
 	}
-};
 
-const registerClickListeners = () => {
+	private hasStateChanged(): boolean {
+		const currentState = this.checkboxes.map(
+			( checkbox ) => checkbox.checked
+		);
+
+		return ! this.initialState.every(
+			( value, index ) => value === currentState[ index ]
+		);
+	}
+
+	private close() {
+		this.dropdown.setAttribute( 'aria-expanded', 'false' );
+
+		if ( this.hasStateChanged() && isReducedMotion() ) {
+			submitForm();
+		}
+	}
+
+	private toggleDropdown() {
+		const isAriaExpanded =
+			this.dropdown.getAttribute( 'aria-expanded' ) === 'true';
+
+		if ( ! isAriaExpanded ) {
+			this.recordInitialState();
+		}
+
+		this.dropdown.setAttribute(
+			'aria-expanded',
+			isAriaExpanded ? 'false' : 'true'
+		);
+
+		if ( isAriaExpanded && this.hasStateChanged() && isReducedMotion() ) {
+			submitForm();
+		}
```

Comment:
```suggestion
		if ( isAriaExpanded && ! isReducedMotion() && this.hasStateChanged() ) {
			submitForm();
		}
```
I think we should submit the form if the user has no reduced motion, right? If so, I would move the condition to the left because it's easier to calculate than all checkbox iterations, and we can/should cache it.

Reply:
Thanks, the form submission should only be triggered for users with the 'Reduced Motion' option enabled. For users who don't have this option enabled, the form is submitted when the checkbox state changes.

Reply:
Oh, got it :smile: